manifests/init.pp

```diff
@@ -307,6 +310,7 @@
   Boolean $jobs_service_enable = $::foreman::params::jobs_service_enable,
   Boolean $hsts_enabled = $::foreman::params::hsts_enabled,
   Array[Stdlib::HTTPUrl] $cors_domains = $::foreman::params::cors_domains,
+  Hash $rails_cache_store = $::foreman::params::rails_cache_store,
```

Comment:
Are the options for cache store well defined? I know from user issue reports that hashes are seen as complex and painful to send on the commandline when configuring through something like `foreman-installer`. If possible, breaking this out would be nice.

Reply:
I fear it is too flexible. Minimum are the following lines in the config which could also be omitted:
```
:rails_cache_store:
  :type: file
```
For redis it would be something like this as minimum:
```
:rails_cache_store:
  :type: redis
  :urls:
     - redis://localhost:8479/0
  :options:
    :compress: true
    :namespace: foreman
```
But it could be multiple URLs and more options.

In Puppet code it would look like:
```
{'type' => 'file'}
```
or:
```
{'type' => 'redis', 'urls' => ['localhost:8479/0'], options => {'compress' => 'true', 'namespace' => 'foreman'}}
```

Not sure how it would look for the installer. I am also not sure if future will perhaps bring more cache store backends and they will have a different format. 

With this in mind I do not come to a good alternative, best other way I can think of:
- rails_cache_store_type as string
- rails_cache_store_redis_urls as array of strings (only used when rails_cache_store_type equals redis)
- rails_cache_store_redis_options as array of strings which will be split on a delimiter (only used when rails_cache_store_type equals redis)